### PR TITLE
feat(spark): upgrade to spark 3.0.0

### PR DESCRIPTION
**Pull request by lyogev**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,89 +6,99 @@ scmInfo := Some(
...
+testOptions in Test := {
+  CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, scalaMajor)) if scalaMajor >= 12 => Seq(Tests.Argument("-l","com.yotpo.metorikku.tags.UnsupportedInCurrentVersion"))
+    case _ => Seq()
+  }
+}
+
+libraryDependencies ++= {
+  CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, scalaMajor)) if scalaMajor == 11 => Seq("com.databricks" %% "spark-redshift" % "3.0.0-preview1" excludeAll excludeAvro)
```
Comment (lyogev) [inline build.sbt:40]: remove, move to latest spark-redshift-community

File: build.sbt [hunk 2]
```diff
@@ -6,89 +6,99 @@ scmInfo := Some(
...
+  "com.fasterxml.jackson.module" %% "jackson-module-scala" % jacksonVersion.value,
+  "com.fasterxml.jackson.dataformat" % "jackson-dataformat-cbor" % jacksonVersion.value,
+  "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion.value,
+  "com.fasterxml.jackson.core" % "jackson-annotations" % jacksonVersion.value,
+  "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion.value,
+  "com.fasterxml.jackson.dataformat" % "jackson-dataformat-yaml" % jacksonVersion.value,
+  "org.apache.commons" % "commons-text" % "1.8",
+  "org.influxdb" % "influxdb-java" % "2.19",
+  // Wait for https://github.com/spark-redshift-community/spark-redshift/pull/72
+  //  "io.github.spark-redshift-community" %% "spark-redshift" % "4.0.1",
```
Comment (lyogev) [inline build.sbt:78]: fix

File: build.sbt [hunk 3]
```diff
@@ -6,89 +6,99 @@ scmInfo := Some(
...
-  "za.co.absa" % "abris_2.11" % "3.2.1"  % "provided" excludeAll(excludeAvro, excludeSpark),
-  "org.apache.hudi" %% "hudi-spark-bundle" % "0.5.2-incubating" % "provided" excludeAll excludeFasterXML,
+  "za.co.absa" %% "abris" % "3.2.1"  % "provided" excludeAll(excludeAvro, excludeSpark),
+  "org.apache.hudi" %% "hudi-spark-bundle" % "0.5.3" % "provided",
   "org.apache.parquet" % "parquet-avro" % "1.10.1" % "provided",
-  "org.apache.avro" % "avro" % "1.8.2" % "provided",
-  "org.apache.hive" % "hive-jdbc" % "2.3.3" % "provided" excludeAll(excludeNetty, excludeNettyAll, excludeLog4j, excludeParquet),
-  "org.apache.hadoop" % "hadoop-aws" % "2.7.3" % "provided",
-  "com.amazon.deequ" % "deequ" % "1.0.4" excludeAll(excludeSpark)
+  "com.amazon.deequ" % "deequ" % "1.0.4" excludeAll(excludeSpark, excludeScalanlp),
```
Comment (lyogev) [inline build.sbt:87]: upgrade

File: docker/spark/k8s/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+RUN wget -q https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-s3/${AWS_SDK_VERSION}/aws-java-sdk-s3-${AWS_SDK_VERSION}.jar -P $SPARK_HOME/jars/
+
+RUN wget -q https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_${SCALA_MAJOR_VERSION}/${SPARK_VERSION}/spark-sql-kafka-0-10_${SCALA_MAJOR_VERSION}-${SPARK_VERSION}.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/org/apache/spark/spark-streaming-kafka-0-10-assembly_${SCALA_MAJOR_VERSION}/${SPARK_VERSION}/spark-streaming-kafka-0-10-assembly_${SCALA_MAJOR_VERSION}-${SPARK_VERSION}.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/org/apache/spark/spark-avro_${SCALA_MAJOR_VERSION}/${SPARK_VERSION}/spark-avro_${SCALA_MAJOR_VERSION}-${SPARK_VERSION}.jar -P $SPARK_HOME/jars/
+RUN wget -q https://repo1.maven.org/maven2/org/apache/commons/commons-pool2/2.6.2/commons-pool2-2.6.2.jar -P $SPARK_HOME/jars/
+
+RUN rm -f $SPARK_HOME/jars/httpclient-*.jar && wget -q https://repo1.maven.org/maven2/org/apache/httpcomponents/httpclient/${HTTPCLIENT_VERSION}/httpclient-${HTTPCLIENT_VERSION}.jar -P $SPARK_HOME/jars
```
Comment (lyogev) [inline docker/spark/k8s/Dockerfile:28]: remove

File: docker/spark/standalone/conf/spark-defaults.conf [hunk 5]
```diff
@@ -13,5 +11,4 @@ spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version=2
 spark.port.maxRetries=0
 spark.rdd.compress=true
 spark.serializer=org.apache.spark.serializer.KryoSerializer
-spark.sql.hive.convertMetastoreParquet=false
-spark.sql.legacy.allowCreatingManagedTableUsingNonemptyLocation=true
```
Comment (lyogev) [inline docker/spark/standalone/conf/spark-defaults.conf:17]: revert

File: docker/spark/standalone/conf/spark-defaults.conf [hunk 6]
```diff
@@ -13,5 +11,4 @@ spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version=2
...
\ No newline at end of file
+spark.sql.hive.convertMetastoreParquet=false
```
Comment (lyogev) [inline docker/spark/standalone/conf/spark-defaults.conf:14]: try and remove

File: examples/hudi/manual_hive_sync_no_partitions_config.yaml [hunk 7]
```diff
@@ -12,7 +12,6 @@ output:
     maxVersions: 1
     hiveDB: default
     parallelism: 1
-    hiveJDBCURL: jdbc:hive2://hive:10000/;auth=noSasl
```
Comment (lyogev) [inline examples/hudi/manual_hive_sync_no_partitions_config.yaml:15]: check why this was removed

**Review by RonBarabash — DISMISSED** [review]

**Pull request merged**